src/imagery/i.eodag/i.eodag.py

```diff
@@ -585,12 +671,50 @@ def filter_result(search_result, geometry, **kwargs):
             operator="le", cloudCover=int(cloud_cover)
         )
 
-    search_result = search_result.filter_date(start=start_date, end=end_date)
+    # queryables are formatted as follow:
+    # {'queryable_1' : [(value_1, operator_1), (value_2, operator_2), (value_3, operator_3), ...],
+    #  'queryable_2' : [(value_1, operator_1), (value_2, operator_2), (value_3, operator_3), ...],
+    #  'queryable_3' : [(value_1, operator_1), (value_2, operator_2), (value_3, operator_3), ...],
+    #  ...
+    #  ...
+    # }
+    if queryables:
+        for queryable, values in queryables.items():
+            if queryable in ["start", "end"]:
+                continue
+            tmp_search_result_list = []
+            for value, operator in values:
+                try:
+                    filtered_search_result_list = search_result.filter_property(
+                        operator=operator, **{queryable: value}
+                    ).data
+                    tmp_search_result_list.extend(filtered_search_result_list)
+                except TypeError:
+                    gs.warning(
+                        _(
+                            "Invalid operator <{}> for queryable <{}>\nOperator <{}> will be used instead".format(
+                                operator, key, DEFAULT_OPERATOR
+                            )
+                        )
+                    )
+                    filtered_search_result_list = search_result.filter_property(
+                        operator=DEFAULT_OPERATOR, **{key: value}
+                    ).data
+                    tmp_search_result_list.extend(filtered_search_result_list)
+            search_result = SearchResult(tmp_search_result_list)
+
+    # Remove duplictes that might be created while filtering
     search_result = remove_duplicates(search_result)
+    if start_date or end_date:
+        search_result = search_result.filter_date(start=start_date, end=end_date)
```

Comment:
Is there a point of filtering by date a second time?

Reply:
If you save a search result to a GeoJSON file... you could then filter these products later by passing the GeoJSON to i.eodag along with the filters (including the date).

Though, when you are searching for products normally, there isn't a need to do so... so filtering again here is to keep it abstracted.

Reply:
Yeah, this is how it works with eodag, first search (which will probably be enough in most cases) and then filter within search results. 

Within GRASS, the filtering could also be done with SQL (in the GUI or with v.db.select or db.* tools) if the user saves the search results as footprints. The attr table of the footprint vector map should contain all queryables.